augur/filter/include_exclude_rules.py

```diff
@@ -898,8 +898,32 @@ def extract_variables(pandas_query: str):
     # covered by Python (unlikely, but I'm not sure). In those cases, don't
     # return anything.
     try:
-        return set(node.id
-                   for node in ast.walk(ast.parse(pandas_query))
-                   if isinstance(node, ast.Name))
+        modified_query, replacement_pair = _replace_backtick_quoting(pandas_query)
+        variables = set(node.id
+                        for node in ast.walk(ast.parse(modified_query))
+                        if isinstance(node, ast.Name))
+        for original_name, generated_name in replacement_pair:
+            variables.remove(generated_name)
+            variables.add(original_name)
+        return variables
     except:
         return None
+
+
+def _replace_backtick_quoting(pandas_query: str):
+    """Replace backtick-quoted names with a generated name.
+
+    Return the modified query and a list of two-tuples representing the original
+    and generated names.
+    """
+    pattern = r"`([^`]+)`"
+    replacement_pairs = []
+
+    def replace(match):
+        original_name = match.group(1)
+        generated_name = f'__augur_filter_{hash(match)}'
```

Comment:
`hash()` isn't what we want here. It can return negative integers, which would make this do the wrong thing.

The generated name can simply use a number that increments.

Reply:
Also, it's not required for our current usage, but I'd think it'd be less surprising to have all instances of the same backtick-quoted name replaced by the same generated name. This suggests making `replacement_pairs` a dict instead of a list.

Reply:
[Updated](https://github.com/nextstrain/augur/compare/7247039abda163fdeaad8975cc7cf4fbd881de69..8d85ffbb18e84d8ed518a0d11cc741a2139827f3) to use a number that increments and a dict.